logic/resolvable/Unifier.java

```diff
@@ -236,6 +248,14 @@ public void addLabelType(Identifier.Variable.Label source, Set<Label> allowedTyp
             unifiedRequirements.types(target, allowedTypes);
         }
 
+        void addConstantValueRequirements(Set<ValueConstraint<?>> values,
+                                          Retrievable id, Retrievable unifiedId) {
+            iterate(values).filter(v -> v.predicate().equals(EQ)).forEachRemaining(value -> {
```

Comment:
definitely need to filter out variables: variables on the RHS cannot be part of the unification process because they would not unify with a variable in the `conclusion`. We can handle constants on the RHS because they can be evaluated during the forward/reverse unification

Reply:
we should delete this `EQ` filter, since the code can already handle all predicates with a constant on the RHS.

This means we've extended concludables to handle all constant predicates (compared to just `=` in the past). As such, we should also make sure the `Retrievable` does not contain these predicates when they are included in a `Concludable`